homeassistant/components/remootio/config_flow.py

```diff
@@ -0,0 +1,179 @@
+_LOGGER = logging.getLogger(__name__)
+
+INPUT_VALIDATION_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, msg="Host is required"): vol.All(
+            vol.Coerce(str),
+            vol.Match(CONNECTION_OPTION_REGEX_HOST),
+            msg="Host appears to be invalid; it can be an IP address or a host name that complies with RFC-1123",
+        ),
+        vol.Required(CONF_API_SECRET_KEY, msg="API Secret Key is required"): vol.All(
+            vol.Coerce(str),
+            vol.Upper,
+            vol.Match(CONNECTION_OPTION_REGEX_API_SECRET_KEY),
+            msg="API Secret Key appears to be invalid; it must be a sequence of 64 characters and can contain only numbers and english letters",
+        ),
+        vol.Required(CONF_API_AUTH_KEY, msg="API Auth Key is required"): vol.All(
+            vol.Coerce(str),
+            vol.Upper,
+            vol.Match(CONNECTION_OPTION_REGEX_API_AUTH_KEY),
+            msg="API Auth Key appears to be invalid; it must be a sequence of 64 characters and can contain only numbers and english letters",
+        ),
+        vol.Required(
+            CONF_DEVICE_CLASS,
+            default=CoverDeviceClass.GARAGE,
+            msg="Controlled device's class is required",
+        ): vol.All(
+            vol.Coerce(str),
+            vol.In([CoverDeviceClass.GARAGE, CoverDeviceClass.GATE]),
+            msg="Controlled device's class appears to be invalid",
+        ),
+    },
+    extra=REMOVE_EXTRA,
+)
+
+DEVICE_NAME = "Remootio Device"
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input."""
+
+    _LOGGER.debug("Validating input... Input [%s]", data)
+
+    data = INPUT_VALIDATION_SCHEMA(data)
+
+    connection_options: ConnectionOptions = ConnectionOptions(
+        data[CONF_HOST], data[CONF_API_SECRET_KEY], data[CONF_API_AUTH_KEY]
+    )
+
+    device_serial_number: str = await get_serial_number(
+        hass, connection_options, _LOGGER
+    )
+
+    data[CONF_SERIAL_NUMBER] = device_serial_number
+
+    return {
+        CONF_TITLE: f"{DEVICE_NAME} (Host: {data[CONF_HOST]}, S/N: {device_serial_number})",
+        CONF_DATA: data,
+    }
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Remootio."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user step."""
+        user_input = user_input or {}
+        errors = {}
+
+        if len(user_input) != 0:
+            validation_result = {}
+
+            try:
+                validation_result = await validate_input(self.hass, user_input)
+            except UnsupportedRemootioDeviceError:
+                _LOGGER.debug("Remootio device isn't supported", exc_info=True)
+                return self.async_abort(reason="unsupported_device")
+            except vol.MultipleInvalid as ex:
+                _LOGGER.error(
```

Comment:
There's no reason why the user should see this in their log, please lower severity to debug
```suggestion
                _LOGGER.debug(
```